### PR TITLE
fix attribute forwarding bug again.

### PR DESCRIPTION
**Pull request by qiwzhang**
It was reverted as suspected to cause crash.  Now crash is fixed,  put the change back.

### CODE REVIEW

File: src/envoy/mixer/mixer_control.h [hunk 1]
```diff
@@ -55,38 +56,37 @@ class MixerControl final : public ThreadLocal::ThreadLocalObject,
...
-                  int check_status_code);
-
-  // Make mixer check call for Tcp requests.
-  void CheckTcp(HttpRequestDataPtr request_data,
-                Network::Connection& connection, std::string origin_user,
-                ::istio::mixer_client::DoneFunc on_done);
+  // Build check request attributes for HTTP.
+  void BuildHttpCheck(HttpRequestDataPtr request_data, HeaderMap& headers,
+                      const ::istio::proxy::mixer::StringMap& map_pb,
+                      std::string origin_user,
```
Comment (lizan) [inline src/envoy/mixer/mixer_control.h]: const reference?

File: src/envoy/mixer/http_filter.cc [hunk 2]
```diff
@@ -230,11 +229,28 @@ class Instance : public Http::StreamDecoderFilter,
       origin_user = ssl->uriSanPeerCertificate();
     }
 
+    // Extract attributes from x-istio-attributes header
+    ::istio::proxy::mixer::StringMap forwarded_attributes;
+    const HeaderEntry* entry = headers.get(Utils::kIstioAttributeHeader);
+    if (entry) {
+      std::string str(entry->value().c_str(), entry->value().size());
```
Comment (lizan) [inline src/envoy/mixer/http_filter.cc]: const

Comment (qiwzhang) [reply src/envoy/mixer/http_filter.cc]: Done

File: src/envoy/mixer/mixer_control.h (hunk 1 above)
Comment (qiwzhang) [reply src/envoy/mixer/mixer_control.h]: Done

**Pull request merged**